dist/sharect.babel.js

```diff
@@ -103,16 +125,25 @@ var Sharect = function () {
 
       div.appendChild(arrow);
 
-      document.body.appendChild(div);
+      if (hasSelection()) {
+        offsetParent(_selection.anchorNode.parentElement).appendChild(div);
+      } else {
+        document.body.appendChild(div);
+      }
     }
 
     function attachEvents() {
-      function hasSelection() {
-        return !!window.getSelection().toString();
-      }
 
       function hasTooltipDrawn() {
-        return !!document.querySelector('.sharect');
+        var tooltip = document.querySelector('.sharect');
+        if (!tooltip) return false;
+
+        if (hasSelection() && offsetParent(tooltip) === offsetParent(_selection.anchorNode.parentElement)) {
+          return true;
+        }
+
+        tooltip.remove();
```

Comment:
If the selection is in a part of the page that has different offsetParent(), kill the tooltip and recreate.